### PR TITLE
Refactor checkout workflow to set shipping address/get shipping methods with graphql

### PR DESCRIPTION
**Pull request by sirugh**
## Description

Uses graphQL to set shipping address and, in the same mutation, receive the available shipping methods for the selected address.

Also removes email from forms for authed users as the email is already attached to cart.
For guest users it will set the guest email on the cart when submitting the address form.

## Related Issue

Closes [PWA-145](https://jira.corp.magento.com/browse/PWA-145).

### Verification Steps

1. Go through checkout -- it should complete successfully. Verify that the REST submissions still work.

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: packages/peregrine/lib/store/reducers/checkout.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ const initialState = {
     receipt: {
         order: {}
     },
-    shippingAddress: null,
+    shippingAddress: {},
```
Comment (sirugh) [inline packages/peregrine/lib/store/reducers/checkout.js]: I may be able to revert this change, though we probably can default this to `{}` safely as it is set to that in `beginCheckout`.

File: packages/venia-ui/lib/components/Checkout/overview.js [hunk 2]
```diff
@@ -57,11 +56,7 @@ const Overview = props => {
                     onClick={handleAddressFormClick}
                     showEditIcon={hasShippingAddress}
                 >
-                    <ShippingAddressSummary
-                        classes={classes}
-                        hasShippingAddress={hasShippingAddress}
-                        shippingAddress={shippingAddress}
```
Comment (sirugh) [inline packages/venia-ui/lib/components/Checkout/overview.js]: It may be overkill to have the summaries import this through context. Can probably at least do away with the threading down to the overview and to the editable forms.

File: packages/venia-ui/lib/components/Checkout/__tests__/__snapshots__/addressForm.spec.js.snap [hunk 3]
```diff
@@ -422,9 +422,7 @@ exports[`renders validation block with message if address is incorrect 1`] = `
     </h2>
     <div
       className="validationMessage"
-    >
-      Oops
-    </div>
+    />
```
Comment (supernova-at) [inline packages/venia-ui/lib/components/Checkout/__tests__/__snapshots__/addressForm.spec.js.snap]: I think this is a legit regression. An error message should be rendered.

File: packages/venia-ui/lib/components/Checkout/addressForm.js [hunk 4]
```diff
@@ -28,16 +31,24 @@ const fields = [
 ];
 
 const AddressForm = props => {
-    const { countries, error, isSubmitting, onCancel, onSubmit } = props;
```
Comment (supernova-at) [inline packages/venia-ui/lib/components/Checkout/addressForm.js]: Okay, this explains the snapshot discrepancy from earlier. Seems like this component no longer takes an `error` prop; the talon provides one instead.

Therefore I think we should:
  1. Remove `error` from this component's `propTypes`
  1. Fix up or remove the "renders validation block with message if address is incorrect" test in `addressForm.spec.js`. It's passing in an `error` prop right now. Perhaps we mock the talon's response instead.
  1. Update the snapshot again to ensure we're rendering an error message

File: packages/venia-ui/lib/components/Checkout/form.css [hunk 5]
```diff
@@ -16,6 +16,12 @@
     text-transform: capitalize;
 }
 
+.disabledPrompt {
+    composes: informationPrompt;
+    /* TODO: Is this the best way to signify disabled? Contrast seems light on text. */
+    color: rgb(var(--venia-grey-dark));
```
Comment (supernova-at) [inline packages/venia-ui/lib/components/Checkout/form.css]: I'd agree maybe we can improve the contrast but since this is consistent with the handful of other "disabled" styles we use in Venia today, I'm 👍 right now.

File: packages/venia-ui/lib/components/Checkout/shippingAddressSummary.js [hunk 6]
```diff
@@ -1,8 +1,16 @@
 import React, { Fragment } from 'react';
 import { array, bool, shape, string } from 'prop-types';
 
+import { useCheckoutContext } from '@magento/peregrine/lib/context/checkout';
+
+import isObjectEmpty from '../../util/isObjectEmpty';
```
Comment (supernova-at) [inline packages/venia-ui/lib/components/Checkout/shippingAddressSummary.js]: In another place we imported this from Peregrine - do we want to do that here too?

File: packages/venia-ui/lib/components/Checkout/shippingMethodSummary.js [hunk 7]
```diff
@@ -1,15 +1,28 @@
 import React, { Fragment } from 'react';
-import { bool, shape, string } from 'prop-types';
+import { shape, string } from 'prop-types';
+
+import { useCheckoutContext } from '@magento/peregrine/lib/context/checkout';
+
+import isObjectEmpty from '../../util/isObjectEmpty';
```
Comment (supernova-at) [inline packages/venia-ui/lib/components/Checkout/shippingMethodSummary.js]: Can import from Peregrine

File: packages/venia-ui/lib/components/Checkout/shippingMethodSummary.js [hunk 8]
```diff
@@ -1,15 +1,28 @@
...
 
 const ShippingMethodSummary = props => {
-    const { classes, hasShippingMethod, shippingTitle } = props;
+    const { classes } = props;
+
+    const [
+        { shippingAddress, shippingMethod, shippingTitle }
+    ] = useCheckoutContext();
```
Comment (supernova-at) [inline packages/venia-ui/lib/components/Checkout/shippingMethodSummary.js]: 💯

File: packages/venia-ui/lib/components/Checkout/form.css (hunk 5 above)
Comment (sirugh) [reply packages/venia-ui/lib/components/Checkout/form.css]: Ok I'll remove the TODO then :)

**Comment by dpatil-magento** [comment]
QA pass but waiting for https://github.com/magento/pwa-studio/pull/2029 to get merged first.
